pkg/apis/crd/v1alpha1/types.go

```diff
@@ -427,3 +354,132 @@ type BGPPeer struct {
 	// a restart before deleting stale routes. The range of the value is from 1 to 3600, and the default value is 120.
 	GracefulRestartTimeSeconds *int32 `json:"gracefulRestartTimeSeconds,omitempty"`
 }
+
+// Source describes the source spec of the packetcapture.
+type Source struct {
+	// Namespace is the source namespace.
+	Namespace string `json:"namespace"`
+	// Pod is the source pod.
+	Pod string `json:"pod"`
+	// IP is the source IPv4 or IPv6 address.
+	IP *string `json:"ip,omitempty"`
+}
+
+// Destination describes the destination spec of the packetcapture.
+type Destination struct {
+	// Namespace is the destination namespace.
+	Namespace string `json:"namespace"`
+	// Pod is the destination pod, exclusive with destination service.
+	Pod string `json:"pod"`
+	// Service is the destination service, exclusive with destination pod.
+	Service *string `json:"service,omitempty"`
+	// IP is the destination IPv4 or IPv6 address.
+	IP *string `json:"ip,omitempty"`
+}
+
+// IPv6Header describes spec of an IPv6 header.
+type IPv6Header struct {
+	// NextHeader is the IPv6 protocol.
+	NextHeader *int32 `json:"nextHeader,omitempty" yaml:"nextHeader,omitempty"`
+	// HopLimit is the IPv6 Hop Limit.
+	HopLimit *int32 `json:"hopLimit,omitempty" yaml:"hopLimit,omitempty"`
+}
+
+// TransportHeader describes spec of a TransportHeader.
+type TransportHeader struct {
+	ICMP *ICMPEchoRequestHeader `json:"icmp,omitempty" yaml:"icmp,omitempty"`
+	UDP  *UDPHeader             `json:"udp,omitempty" yaml:"udp,omitempty"`
+	TCP  *TCPHeader             `json:"tcp,omitempty" yaml:"tcp,omitempty"`
+}
+
+// ICMPEchoRequestHeader describes spec of an ICMP echo request header.
+type ICMPEchoRequestHeader struct {
+}
+
+// UDPHeader describes spec of a UDP header.
+type UDPHeader struct {
+	// SrcPort is the source port.
+	SrcPort *int32 `json:"srcPort,omitempty"`
+	// DstPort is the destination port.
+	DstPort *int32 `json:"dstPort,omitempty"`
+}
+
+// TCPHeader describes spec of a TCP header.
+type TCPHeader struct {
+	// SrcPort is the source port.
+	SrcPort *int32 `json:"srcPort,omitempty"`
+	// DstPort is the destination port.
+	DstPort *int32 `json:"dstPort,omitempty"`
```

Comment:
I know I suggested pointers for this originally. Given more recent discussions, I don't really feel strongly about it anymore, so given that 0 is not a valid port number typically, I would be ok with removing the pointers. I'll let @tnqn chime in as well.

Reply:
temporary reset to int32 type.

Reply:
My first impression is this should have a pointer because it's optional. `if header.SrcPort != nil` is the condition in which we should capture specifc source port only, right?

Reply:
yes, it's optional. sounds more reasonable to set it as pointer.